server/src/test/java/org/elasticsearch/search/suggest/CustomSuggesterSearchIT.java

```diff
@@ -135,7 +149,7 @@ protected XContentBuilder innerToXContent(XContentBuilder builder, Params params
 
         @Override
         public String getWriteableName() {
-            return "custom";
+            return SUGGESTION_NAME;
         }
 
         @Override
```

Comment:
Is it possible to move this test and the custom suggestion impl in the plugin examples and adds rest tests that checks that xcontent parsing works ? I am not sure that this IT test is enough to validate the plugin integration.
This would be nice to have a real example that shows how to write a custom suggester and the example plugins are the perfect fit for this.

Reply:
I agree on both, that this doesn't completely validate the plugin integration, and that it would be preferable to have it in an example plugin. I was going to make that a follow up PR but I'll add it to this one